Documentation/proposals/202405-docs-restructuring.md

```diff
@@ -0,0 +1,104 @@
+
+# How
+
+I believe a basic structure for the website should be as given below:
+
+### 1. Getting Started
+
+- **Introduction** - This section will introduce us to Prometheus-Operator and will talk about goals and the problems that are being solved by this project.
+- **Installation Guide** - This section will contain all the methods of installation of Prometheus-Operator in the production environment.
+- **Compatibility** - This section will provide information about compatibility of Prometheus-Operator with Kubernetes, Prometheus, Alertmanager and Thanos.
+- **Design** - This section will describe the design of Prometheus-Operator focusing on the various custom resource definitions(CRDs) it manages.
+
+### 2. API Reference
+
+This section will provide detailed information about different fields of the Custom Resources in Prometheus-Operator(config, spec, status and other information).
+
+### 3. User Guide
+
+This section will provide focused information for developers who want to integrate Prometheus and Alertmanager using Prometheus-Operator in their applications along with examples and troubleshooting.
+
+### 4. Administrator Guide
+
+This section will provide deeper insights for administrators who want to set up a robust, secure monitoring infrastructure.
+
+### 5. Community
+
+- **Contributing**
+- **Testing**
+- **DCO**
+- **License**
+- **Code of Conduct**
+- **Release**
+- **Adopters**
+
+The reasoning behind the new structure is given below -
+
+- The **“Getting-Started”** section should only serve information on how to begin your journey with the project. So, the first thing that comes to mind should be a basic overview followed by the installation. It won’t make sense to have the installation section appear in the User-Guide(in the current model) because installation is the foremost important thing to begin working on a project. With installation, users would also like to know the different releases and their compatibility, thus it makes more sense to add this topic just below the installation section. At last, Design doc should be present to give an overview of how different components interact and what their purpose is in a concise manner.
+
+- API Reference is of extreme importance and is something needed by developers to implement the functionalities they need. It is something needed by both a **User** and an **Administrator**. As it is common for both personas, I think rather than duplicating this page in both sections, we should have a separate section for it due to its relevance. Further, looking at the current API reference page, it contains too much information for a single page. This makes the page look overly crowded and searching for any particular field becomes a tedious task. We can convert this page into multiple pages by grouping sections and cross-reference properly which will provide a better experience for a user.
```

Comment:
I could see 1 page per CRD.

Reply:
I think we can first categorise by different packages then we can move on to different pages for different CRDs for each package.